File: EX3/soap/server/src/main/java/hello/DummyData.java
package hello;

import jakarta.xml.bind.annotation.XmlElement;

public class DummyData {
    @XmlElement
    String mFirstName;
    @XmlElement
    String mLastName;
    public DummyData() {}
    public DummyData(String _FirstName, String _lastName) {
        mFirstName = _FirstName;
        mLastName = _lastName;
    }
    public String toString() {
        return "hello.DummyData from (" + mLastName + ") --> " + mFirstName;
    }
}

File: EX3/lib/src/main/kotlin/classes/DummyData.kt
package classes

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "dummy")
@XmlAccessorType(XmlAccessType.FIELD)
data class DummyData(
    @field:XmlElement(name = "first-name")
    val mFirstName: String,
    @field:XmlElement(name = "last-name")
    val mLastName: String
) {
    constructor() : this("", "")
}